src/backend/src/services/slack.services.ts

```diff
@@ -0,0 +1,289 @@
+        }
+        return 'emoji:' + block.name;
+      case 'link':
+        if (block.text) {
+          return `${block.text}:(${block.url})`;
+        }
+        return block.url ?? '';
+      case 'text':
+        return block.text ?? '';
+      case 'user':
+        //users are represented as an id, get the name of the user from the slack api
+        let userName: string = block.user_id ?? '';
+        try {
+          userName = (await getUserName(block.user_id ?? '')) ?? `Unknown User:${block.user_id}`;
+        } catch (error) {
+          userName = `Unknown_User:${block.user_id}`;
+        }
+        return '@' + userName;
+      case 'usergroup':
+        return `usergroup:${block.usergroup_id}`;
+    }
+  }
+
+  /**
+   * Gets the users notified in a specific SlackRichTextBlock.
+   * @param block the block that may contain mentioned user/users
+   * @param usersSettings the settings of all the users in prisma
+   * @param channelId the id of the channel that the block is being sent in
+   * @returns an array of prisma user ids of users to be notified
+   */
+  private static async blockToMentionedUsers(
+    block: SlackRichTextBlock,
+    usersSettings: User_Settings[],
+    channelId: string
+  ): Promise<string[]> {
+    switch (block.type) {
+      case 'broadcast':
+        switch (block.range) {
+          case 'everyone':
+            return usersSettings.map((usersSettings) => usersSettings.userId);
+          case 'channel':
+          case 'here':
+            //@here behaves the same as @channel; notifies all the users in that channel
+            let slackIds: string[] = [];
+            try {
+              slackIds = await getUsersInChannel(channelId);
+            } catch (ignored) {}
+            return usersSettings
+              .filter((userSettings) => {
+                return slackIds.some((slackId) => slackId === userSettings.slackId);
+              })
+              .map((user) => user.userId);
+          default:
+            return [];
+        }
+      case 'user':
+        return usersSettings
+          .filter((userSettings) => userSettings.slackId === block.user_id)
+          .map((userSettings) => userSettings.userId);
+      default:
+        //only broadcasts and specific user mentions add recievers to announcements
+        return [];
+    }
+  }
+
+  /**
+   * Given a slack event representing a message in a channel,
+   * make the appropriate announcement change in prisma.
+   * @param event the slack event that will be processed
+   * @param organizationId the id of the organization represented by the slack api
+   * @returns an annoucement if an announcement was processed and created/modified/deleted
+   */
+  static async processMessageSent(event: SlackMessageEvent, organizationId: string): Promise<Announcement | undefined> {
+    let slackChannelName: string;
+    //get the name of the channel from the slack api
+    try {
+      slackChannelName = await getChannelName(event.channel);
```

Comment:
lets make getChannelName nonthrowing, and it returns either the channel name or undefined, then we just null coalesce this value to use unknow_channel as the default